web/src/components/njwds-extended/Heading.tsx

```diff
@@ -0,0 +1,110 @@
+import { ReactElement } from "react";
+
+export type ValidHeadingLevel = 0 | 1 | 2 | 3 | 4;
+export type HeadingStyleVariantName = "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "rawElement";
+
+type HeadingStyles = {
+  [variantName in HeadingStyleVariantName]: string;
+};
+
+const variantClasses: HeadingStyles = {
+  h1: "h1-styling",
+  h2: "h2-styling",
+  h3: "h3-styling",
+  h4: "h4-styling",
+  h5: "h5-styling",
+  h6: "h6-styling",
+  rawElement: "",
+};
+
+const determineClasses = (
```

Comment:
I agree with you here, so I'd love to propose some options:

1. `determineClassNames` - since it determines 0 - many classes
2. `generateClassList` - since it's technically a generator function
3. `generateClassNameString` - the one I like the least but technically the most accurate?

Reply:
I like my P.S. suggestion of `determineClassNames`, but could maybe like (or not) `determineClassList` as opposed to `generateClassList`. Did you have a different #1 choice?

With regards to naming the function `generate...`: 1. I believe all `generate...` functions live in `factories.ts` so we'd have to move it there; 2. If no other component is using this function (or will), it can live in `Heading.tsx` and we don't need to/shouldn't name it `generate...`.

Reply:
This is very valuable feedback. Thank you. I should have noticed that.